libres/lib/analysis/fwd_step_enkf.cpp

```diff
@@ -538,29 +538,23 @@ void *fwd_step_enkf_get_ptr(const void *arg, const char *var_name) {
     }
 }
 
-#ifdef INTERNAL_LINK
-#define LINK_NAME FWD_STEP_ENKF
-#else
-#define LINK_NAME EXTERNAL_MODULE_SYMBOL
-#endif
-
-analysis_table_type LINK_NAME = {.name = "FWD_STEP_ENKF",
-                                 .updateA = fwd_step_enkf_updateA,
-                                 .initX = NULL,
-                                 .init_update = NULL,
-                                 .complete_update = NULL,
-
-                                 .freef = fwd_step_enkf_data_free,
-                                 .alloc = fwd_step_enkf_data_alloc,
-
-                                 .set_int = fwd_step_enkf_set_int,
-                                 .set_double = fwd_step_enkf_set_double,
-                                 .set_bool = fwd_step_enkf_set_bool,
-                                 .set_string = fwd_step_enkf_set_string,
-                                 .get_options = fwd_step_enkf_get_options,
-
-                                 .has_var = fwd_step_enkf_has_var,
-                                 .get_int = fwd_step_enkf_get_int,
-                                 .get_double = fwd_step_enkf_get_double,
-                                 .get_bool = fwd_step_enkf_get_bool,
-                                 .get_ptr = fwd_step_enkf_get_ptr};
+analysis_table_type FWD_STEP_ENKF = {.name = "FWD_STEP_ENKF",
+                                     .updateA = fwd_step_enkf_updateA,
+                                     .initX = NULL,
+                                     .init_update = NULL,
+                                     .complete_update = NULL,
+
+                                     .freef = fwd_step_enkf_data_free,
+                                     .alloc = fwd_step_enkf_data_alloc,
+
+                                     .set_int = fwd_step_enkf_set_int,
+                                     .set_double = fwd_step_enkf_set_double,
+                                     .set_bool = fwd_step_enkf_set_bool,
```

Comment:
Outside of scope, I know, but I think turning `analysis_table_type` into a cpp class should be reasonably simple and give you the bang for your buck of letting the analysis modules `FwdStepEnkf`, `BootStrapEnkf`, `IESEnkf` and `StdDebugEnkf` inherit and override from an abstract `AnalysisModule` class rather than initializing the `analysis_table_type` struct and shipping that around.

Reply:
This is somewhat overlapping with the work done by Sloths currently so perhaps some coordination is due @oyvindeide?